### ISSUE TITLE
Fix .item() usage when scaling Atoms in _create_product.

### ISSUE THREAD

**Issue by zfrangella** [udellgroup/rlaopt#142]
In _create_product in utils.py we do the following to form the product of a constant and an atom:

```
 # Handle atom scaling: const * atom
                from rlaopt.atoms.atom import Atom

                if isinstance(other, Atom):
                    scalar = folded_const.value.item()
                    scaled = other._scale(scalar)
                    if scaled is not NotImplemented:
                        return scaled
```

As we are using .item() to get scalar,  the computational graph can end up breaking leading to trouble when we go to use autograd, particularly when we want to differentiate through the solver, a shown in the following example:

```
def obj(mu: torch.Tensor) -> torch.Tensor:
    x = Variable((p,), name="x")
    train_obj = 1 / ntr * SumSquares(Atr @ x - btr) + mu * L1Norm(x)

    config = ProxGradConfig(eta=0.5, use_linesearch=True)
    opt = ProxGrad(train_obj, config, detach=False)
    result = opt.solve()

    x_sol = result.variable_values
    
    return 1 / ntst * torch.linalg.norm(Atst @ x_sol['x'] - btst) ** 2
```

If we go to differentiate the solver:

```
mu0 = torch.tensor(1.0, requires_grad=True)
grad = torch.func.grad(obj)(mu0)
print(grad)
```

The print statement shows the incorrect value 0.0, as the gradient information in mu0 has been lost.
Note if we had instead used L1Norm(x, mu), we would have obtained the correct answer as this approach does not use _create_product.
